### DISCUSSION TITLE
Trying to run sh script by keybind.

### DISCUSSION THREAD

**Discussion by ghost** [qtile/qtile#3451] [Q&A] [answered]
Hey, everyone! I'm trying to run rofi using a keybind. I have tried this so far (reading [this](https://www.mail-archive.com/qtile-dev@googlegroups.com/msg01883.html) disscussion)
1.```Key[mod[, "p", qtile.cmd_spawn("home/user/path/to/sh"))```
2.```Key[mod[, "p", qtile.cmd_spawn(os.path.expanduser("home/user/path/to/sh")))```
But none of them works. I also tried using `~` insdead of `/home/user/`, and nothing changed.

Also after I restart qtile rofi window opens, but keybinding is not working after that. 
Thanks in advance for help!

**Comment by m-col** [comment]
I think you want something like this:
```py
Key(mods, "p", lazy.spawn("~/path/to/sh", shell=True))
```
Calling `qtile.cmd_spawn("home/user/path/to/sh")` directly like that runs that command _at that time_ i.e. when the config is loaded, which is why you're seeing rofi at startup and not later. The key binding that is created is actually being bound to the return of that function, which I'd guess is just `None`. Using the `lazy` version like above describes what command you'd like to evaluate at a later time i.e. when the key is pressed.

**Reply by ghost** [reply]
Now rofi doesn't launch after restart, but keybinding still doesn't work.

**Reply by m-col** [reply]
Do you see any log output in ~/.local/share/qtile/qtile.log when you press the key combo?

**Reply by ghost** [reply]
No, nothing changes.

**Comment by SweepingBishops** [comment]
It probably was a typo, but just saying that it should be
```Key([mod],...```
instead of
```Key[mod[,...```

 I have a small function in my config for this.
```
def command_run(qtile,command):
    os.system(command)
```
Then when defining the Keys you can pass in the command you want to run in the shell. Eg:
```
Key([mod], 'p', lazy.function(command_run,'~/path/to/sh'))
```
You'll also want to put this line with the other imports at the top of the config:
```
import os.system
```

**Comment by hamensman** [comment, marked as answer]
@dkoselevs Assuming you're talking about running _this_ rofi build (https://github.com/adi1090x/rofi) - which I myself use - I have this:
```python
Key([mod], "a", lazy.spawn("/home/user/.config/rofi/bin/launcher_misc")),
Key([mod, "shift"], "q", lazy.spawn("/home/user/.config/rofi/bin/powermenu"), desc="Launch power menu"),
```
Obviously make sure the scripts are already executable.
